front/app/modules/commercial/insights/admin/containers/Insights/Edit/InputsTable/InputsTableRow.tsx

```diff
@@ -25,41 +25,44 @@ const CategoryList = styled.div`
 
 type InputsTableRowProps = {
   input: IInsightsInputData;
-  onSelect: () => void;
+  selected: boolean;
+  changeSelected: () => void;
+  onPreview: () => void;
 };
 
-const InputsTableRow = ({ input, onSelect }: InputsTableRowProps) => {
+const InputsTableRow = ({
+  input,
+  selected,
+  changeSelected,
+  onPreview,
+}: InputsTableRowProps) => {
   const idea = useIdea({ ideaId: input.relationships?.source.data.id });
 
   if (isNilOrError(idea)) {
     return null;
   }
 
-  // TODO: Implement checkbox logic
-  const handleCheckboxChange = () => {};
-
   const handleEnterPress = (
     event: React.KeyboardEvent<HTMLTableRowElement>
   ) => {
     if (event.key === 'Enter') {
-      onSelect();
+      onPreview();
     }
   };
 
   return (
     <tr
       data-testid="insightsInputsTableRow"
-      onClick={onSelect}
       tabIndex={0}
       onKeyPress={handleEnterPress}
     >
       <td>
-        <Checkbox checked={false} onChange={handleCheckboxChange} />
+        <Checkbox checked={selected} onChange={changeSelected} />
       </td>
-      <td>
+      <td onClick={onPreview}>
```

Comment:
I see why this was needed but can we fix it with stopPropagation() on the click event for the checkbox? It would be better if the entire row is clickable, I think.

Reply:
I had to make a change to the checkbox component to achieve this (Since it was relying on the label click, stopping propagation wouldn't work). PR up soon